doc/source/google/step-zero-gcp.rst

```diff
@@ -102,14 +102,54 @@ your google cloud account.
       kubectl create clusterrolebinding cluster-admin-binding \
         --clusterrole=cluster-admin \
         --user=<GOOGLE-EMAIL-ACCOUNT>
-    
+
    Replace `<GOOGLE-EMAIL-ACCOUNT>` with the exact email of the Google account
    you used to sign up for Google Cloud.
 
    .. note::
-  
+
       Did you enter your email correctly? If not, you can run `kubectl delete
       clusterrolebinding cluster-admin-binding` and do it again.
 
+7. [optional] Create a node pool for users
+
+  This is an optional step, for those who want to separate
+  user pods from "core" pods such as the Hub itself and others.
+  See :doc:`../optimization` for details on using a dedicated user node pool.
+
+  The nodes in this node pool are for the users only. The node pool has
+  autoscaling enabled along with a lower and an upper scaling limit. This
+  means that the amount of nodes is automatically adjusted along with the
+  amount of users scheduled.
+
+  The `n1-standard-2` machine type has 2 CPUs and 7.5 GB of RAM each of which
+  about 0.2 CPU will be requested by system pods. It is a suitable choice for a
+  free account that has a limit on a total of 8 CPU cores.
+
+  Note that the node pool is *tainted*. Only user pods that are configured
+  with a *toleration* for this taint can schedule on the node pool's nodes.
+  This is done in order to ensure the autoscaler will be able to scale down
+  when the user pods have stopped.
+
+  .. code-block:: bash
+
+    gcloud beta container node-pools create user-pool \
+      --machine-type n1-standard-2 \
+      --num-nodes 0 \
+      --enable-autoscaling \
+      --min-nodes 0 \
+      --max-nodes 3 \
+      --node-labels hub.jupyter.org/node-purpose=user \
+      --node-taints hub.jupyter.org_dedicated=user:NoSchedule
+
+  .. note::
+
+    Consider adding the ``--preemptible`` flag to reduce the cost
+    significantly. You can `compare the prices here
+    <https://cloud.google.com/compute/docs/machine-types>`_. See
+    the `preemptible node documentation
+    <https://cloud.google.com/compute/docs/instances/preemptible>`_ for more
+    information.
```

Comment:
Related to this, is an issue that is not yet resolved I think: https://github.com/jupyterhub/kubespawner/issues/223

Reply:
In summary, I recall that whenever a user is evicted, as they are from preemptible node pools that run out of their usage time for some reason, they will not be able to login again without a hub-pod restart.

Reply:
I don't think that's something we should look at for this release. Do you think this means we should not recommend looking at preemptible nodes for now?

Reply:
commented out the recommendation for now

Reply:
Yeah, I would not recommend them until one of us have successfully utilized them with users that authenticate regularly etc. I ended up with trouble doing it.